src/AbstractCas.php

```diff
@@ -152,10 +152,6 @@ public function requestTicketValidation(
         array $parameters = [],
         ?ResponseInterface $response = null
     ): ?ResponseInterface {
-        if (false === $this->supportAuthentication()) {
```

Comment:
This change is out of scope of this PR, maybe we should open another PR and discuss about it ?

Reply:
Sure, we can do that. The reasoning for this removal is to be able to send tickets Authorization headers instead of query strings, but I understand we are bound by the CAS specs, though I believe the limitation is purely theoretical. There's nothing less safe than the querystring no? 

FYI
https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters

Reply:
I guess this will be fixed with the other comment in this PR.

Now, regarding your link and as far as I understand.
I understand Github and why they are doing that, but there is a difference with the CAS protocol. The token which is sent in `ticket` as a parameter in the URL is a one time token. I think there's a difference, security wise, between a one-time token and a permanent access token.